server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -878,6 +878,10 @@ protected GetResult realtimeGetUnderLock(
         }
         boolean getFromSearcherIfNotInTranslog = getFromSearcher;
         if (versionValue != null) {
+            // It is possible that once we've seen the ID in the version map and attempt handling the get,
+            // we won't see the doc in the translog since it might have been moved out. In these cases, we
+            // should always fall back to get the doc from the internal searcher.
```

Comment:
Can we perhaps note in the comment that ideally we should keep the translog around?

Reply:
Can we also add a comment to say this is also for the case when the location is not tracked in the versionMap which is the original use case for this flag? Basically this flag now cover two cases.